client/js/components/main/assessment.spec.js

```diff
@@ -7,108 +7,115 @@ import appHistory              from "../../history";
 import { Assessment }          from "./assessment";
 import * as AssessmentActions  from "../../actions/assessment";
 
-describe("assessment", function() {
-  var props;
-  var allQuestions,
+
+var props;
+var allQuestions,
+  assessment,
+  assessmentViewed,
+  checkedResponses,
+  currentItem,
+  outcomes,
+  previousQuestions,
+  progress,
+  questionCount,
+  questionsPerPage,
+  responses,
+  result,
+  settings,
+  subject,
+  submitAssessment;
+
+var result;
+var subject;
+
+var reset = () => {
+  allQuestions = () => [{
+    timeSpent:0,
+    outcomes:{shortOutcome:"", longOutcome:""},
+    material: "Test Material",
+    correct:[{
+      id:"testCorrect",
+      value:"100"
+    }],
+    points_possible:"1",
+    question_type:"multiple_choice_question",
+    answers:[
+      {
+        id:"testCorrect",
+        material:"test question material"
+      },
+      {
+        id:"testIncorrect",
+        material:"test question material"
+      }
+    ],
+    title:"Test Question Title",
+    id:"TestQuestionID"
+  }];
+
+  assessment = {
+    title: "Test Title"
+  };
+
+  checkedResponses = [];
+
+  currentItem = 5;
+
+  questionsPerPage = 1;
+
+  outcomes = () => {};
+
+  progress = {
+    currentItemIndex:0
+  };
+
+  questionCount = 10;
+
+  responses = [];
+  settings = {
+    user_id      : 0,
+    max_attempts : 1,
+    eid          : "external_identifier",
+    src_url      : "http://www.openassessments.com/api/assessments/55.xml",
+    questions_per_page:1,
+    assessment_kind: "SUMMATIVE"
+  };
+
+  props = {
+    allQuestions:allQuestions(),
     assessment,
-    assessmentViewed,
+    assessmentViewed: () => {},
     checkedResponses,
     currentItem,
-    outcomes,
-    previousQuestions,
+    hideLMSNavigation: () => {},
+    nextQuestions: () => {},
+    outcomes:outcomes(),
+    previousQuestions: () => {},
     progress,
     questionCount,
     questionsPerPage,
     responses,
-    result,
+    scrollParentToTop: () => {},
+    sendSize: () => {},
     settings,
-    subject,
-    submitAssessment;
+    submitAssessment: () => {}
+  };
+
+
+  result = TestUtils.renderIntoDocument(<Assessment {...props} />);
+  subject = ReactDOM.findDOMNode(result);
+};
+
+
+describe("assessment", function() {
+
 
   // Props are reset to these default values between each test. To use
   // a modified prop in your test case, modify props.myProp in your test case and
   // re-render dom element with  result = TestUtils.renderIntoDocument. Then
   // proceed to test your result.
   beforeEach(() => {
-    allQuestions = () => [{
```

Comment:
This was moved outside of the describe block so that nested describe blocks could still access these variables. I wanted to add more describe blocks because it was starting to just be a wall of tests.